cifsiostat.c

```diff
@@ -522,6 +522,11 @@ int main(int argc, char **argv)
 		}
 
 		else if (!strncmp(argv[opt], "--dec=", 6) && (strlen(argv[opt]) == 7)) {
+			/* Check that the argument is a digit */
+			if (!isdigit(argv[opt][6])) {
```

Comment:
@lzaoral I think this assumes only a single digit is given, but input could be something like "1a", "1O" or "10" and still pass through the check here.  If we want to handle this case too, the usual way is to use the end marker from a function like strtol(3) instead of atoi(3).  This can simplify the code a little too as there's just one error handling branch then, e.g.

```
    char *endnum;
    dplaces_nr = strtol(argv[opt] + 6, &endnum, 10);
    if (*endnum != '\0' || dplaces_nr < 0 || dplaces_nr > 2) {
        usage(argv[0]);
    }
```

Reply:
Yes, it assumes only a single digit. Originally, I wanted to use `strtol(3)` but the condition to reach this code already requires only a single character parameter:
```c
!strncmp(argv[opt], "--dec=", 6) && (strlen(argv[opt]) == 7)
```

Reply:
Good point :)   ... the strlen could be removed if you prefer the strtol - but you're right, either way is correct.